releng/raw-binaries/fs/usr/local/bin/influxdb_raw_binaries.bash

```diff
@@ -15,29 +17,62 @@ To build with race detection enabled, pass the -r flag.
 }
 
 RACE_FLAG=""
+STATIC=""
 
-while getopts hr arg; do
+while getopts w:hrs arg; do
   case "$arg" in
     h) printHelp; exit 1;;
     r) RACE_FLAG="-race";;
+    s) STATIC=1;;
   esac
 done
 
-
 if [ -z "$GOOS" ] || [ -z "$GOARCH" ]; then
   >&2 echo 'The environment variables $GOOS and $GOARCH must both be set.'
   exit 1
 fi
 
+if [[ "$GOOS" == darwin ]] ; then
+  # Control the compiler for go. Rust linker is set in $HOME/.cargo/config
+  export CC=x86_64-apple-darwin15-clang
+fi
+
+WORKSPACE=/influxdata
+
+mkdir -p ${WORKSPACE}
 
-# Extract tarball into GOPATH.
-tar xz -C "$GOPATH" -f /influxdb-src.tar.gz
+echo "Extracting influxdb tarball"
+# Extract tarball into WORKSPACE.
 
-SHA=$(jq -r .sha < "$GOPATH/src/github.com/influxdata/influxdb/.metadata.json")
+tar -vxz -C ${WORKSPACE} -f /influxdb-src.tar.gz
 
+SHA=$(jq -r .sha < "${WORKSPACE}/influxdb/.metadata.json")
+
+OUTDIR=$(mktemp -d)
+(
```

Comment:
Why the subshell here?

Reply:
the `cd` happens in the subshell without affecting the working directory outside the sub shell